### DISCUSSION TITLE
How does make() work?

### DISCUSSION THREAD

**Discussion by garma83** [CodeDredd/pinia-orm#1169] [Q&A] [answered]
Hi,

I imagine the purpose of the make() function is to provide a draft of the model, so it can be changed by the user and then persisted to the store and / or backend once the user is ready.

However how should I do this? I can't seem to make changes to the model. I tried this:
let model = useRepo(Model).make()

model.description = 'hi';
useRepo(Model).save({description:'hi'})

neither of these approaches seem to work

Also, when I am ready to persist it how should I do this?

**Comment by CodeDredd** [comment, marked as answer]
HI,

If your primary key is an `uid` field in the model, then you can do this:

````ts
const modelRepo = useRepo(Model)
const model = modelRepo.make()

model.description = 'hi';
modelRepo.save(model);
````

otherwiese you also need to pass the primary key which i think is clear why. :)

**Reply by garma83** [reply]
appreciated, ty!
